ignite/metrics/metric.py

```diff
@@ -46,9 +40,6 @@ def __init__(self, output_transform: Callable = lambda x: x, device: Optional[Un
                     "across all computing devices".format(self.__class__.__name__),
                     RuntimeWarning,
                 )
-            if device is None:
-                device = "cuda"
-            device = torch.device(device)
         self._device = device
```

Comment:
Is `_device` used ?

Reply:
Yes, it is used in certain metrics that proposes to store internals on a device.